spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -311,26 +321,36 @@ function onRecvPacket(packet: Packet) {
        amount: packet.amount
      }
      tokens = []Token{token}
+     receiver = packet.receiver
   } else if transferVersion == "ics20-2" {
     FungibleTokenPacketDataV2 data = UnmarshalJSON(packet.data)
     tokens = data.tokens
+    // if we need to forward the tokens onward
+    // temporarily send to the channel escrow address of the intended receiver
+    if len(forwardingPath) > 0 {
+      receiver = channelForwardingAddresses[packet.destinationChannel]
+    }
   } else {
     // should never be reached as transfer version must be negotiated to be either
     // ics20-1 or ics20-2 during channel handshake
     abortTransactionUnless(false)
   }
 
+  assert(packet.sender !== "")
+  assert(receiver !== "")
```

Comment:
Note that this assertion may fail if we are in the case where `transferVersion == ics-20-2 && forwardingPath is nil` because we are not setting the `receiver` variable. If I got it correctly this would be the case where we are the destChain and we don't need to keep forwarding. Thus we should handle this scenario setting a proper receiver to allow lines 362 or 374 to execute properly.